tests/test_errors.py

```diff
@@ -143,6 +157,18 @@ def test_pydantic_error_type():
     assert repr(e) == "Invalid JSON: Test [type=json_invalid, context={'error': 'Test'}]"
 
 
+def test_pydantic_error_type_nested_ctx():
+    e = PydanticKnownError('json_invalid', {'error': 'Test', 'foo': {'bar': []}})
+    assert e.message() == 'Invalid JSON: Test'
+    assert e.type == 'json_invalid'
+    # TODO fix inconsistency here with context. It should include "foo" key
+    # assert e.context == {'error': 'Test', 'foo': {'bar': []}}
```

Comment:
This seems like a bug right?

Reply:
Yeah it looks like a bug, it requires changing the `pub enum ErrorType` and `extract_context`-macro to include the actual given context also instead of just picking the certain fields there. Should I fix it here also?

Reply:
If you're willing to solve this, it would be greatly appreciated. If it looks complex or not urgent for you I think it's also acceptable for us to file this as an issue and leave for a later PR.

Reply:
Lets get this PR in. I can check out what it requires to fix the inconsistency in separate one